packages/lib/src/toolbar/controls/Selector/Selector.tsx

```diff
@@ -28,54 +37,136 @@ function Selector<T extends string>(props: Props<T>) {
     disabled,
     onChange,
     options,
-    optionComponent: Option,
+    optionComponent: OptionComp,
   } = props;
 
-  const { height: winHeight } = useWindowSize();
-  const menuMaxHeight = winHeight - MENU_TOP - MENU_BOTTOM;
+  const [isOpen, toggle] = useToggle();
+  const [activeIndex, setActiveIndex] = useState<number | null>(null);
+  const [selectedIndex, setSelectedIndex] = useState(() => {
+    return getAllOptions(options).indexOf(value);
+  });
+
+  const labelId = useId();
+  const referenceId = useId();
+  const selectedOptionId = useId();
+  const listRef = useRef<(HTMLButtonElement | null)[]>([]);
+
+  const { refs, floatingStyles, context } = useFloating<HTMLButtonElement>({
+    open: isOpen,
+    middleware: [
+      offset(6),
+      size({
+        padding: 12,
+        apply({ availableHeight, elements, rects }) {
+          Object.assign(elements.floating.style, {
+            maxHeight: `${Math.min(availableHeight, MENU_MAX_HEIGHT)}px`,
+            minWidth: `${rects.reference.width}px`,
+          });
+        },
+      }),
+      shift({ padding: 6 }),
+    ],
+    onOpenChange: toggle,
+    whileElementsMounted: autoUpdate,
+  });
+
+  const { getReferenceProps, getFloatingProps, getItemProps } = useInteractions(
+    [
+      useClick(context),
+      useFloatingDismiss(context),
+      useListNavigation(context, {
+        listRef,
+        activeIndex,
+        loop: true,
+        focusItemOnHover: false,
+        onNavigate: setActiveIndex,
+      }),
+    ],
+  );
+
+  function handleSelect(index: number, option: T) {
+    setSelectedIndex(index);
+    toggle(false);
+    onChange(option);
+  }
 
   return (
     <div className={styles.root}>
-      {label && <span className={styles.selectorLabel}>{label}</span>}
+      {label && (
+        <span id={labelId} className={styles.selectorLabel}>
+          {label}
+        </span>
+      )}
 
-      <Wrapper className={styles.wrapper} onSelection={onChange}>
-        <Button className={styles.btn} tag="button" disabled={disabled}>
-          <div className={styles.btnLike}>
-            <span className={styles.selectedOption}>
-              <Option option={value} isSelected />
-            </span>
-            <MdArrowDropDown className={styles.arrowIcon} />
-          </div>
-        </Button>
+      <button
+        ref={refs.setReference}
+        id={referenceId}
+        className={styles.btn}
+        type="button"
+        disabled={disabled}
+        role="combobox"
+        aria-labelledby={`${label ? labelId : ''} ${selectedOptionId}`}
```

Comment:
Note that I'm combining the label of the selector with the selected option. Floating UI's `useRole` doesn't do this, so I'm glad I didn't use it; I saw this pattern in React Aria.